x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx

```diff
@@ -0,0 +1,196 @@
+  name: string;
+}
+
+// type MappedMetrics = Record<keyof HostNodeRow, SnapshotNodeMetric>;
+
+interface HostTableParams {
+  time: TimeRange;
+}
+
+/**
+ * Helper functions
+ */
+const formatMetric = (type: SnapshotMetricInput['type'], value: number | undefined | null) => {
+  return value || value === 0 ? createInventoryMetricFormatter({ type })(value) : 'N/A';
+};
+
+const buildItemsList = (nodes: SnapshotNode[]) => {
+  return nodes.map(({ metrics, path, name }) => ({
+    name,
+    os: path.at(-1)?.os ?? '-',
+    title: {
+      name,
+      cloudProvider: path.at(-1)?.cloudProvider ?? null,
+    },
+    ...metrics.reduce((data, metric) => {
+      data[metric.name as HostMetric] = metric;
+      return data;
+    }, {} as HostMetrics),
+  })) as HostNodeRow[];
+};
+
+/**
+ * Columns translations
+ */
+const titleLabel = i18n.translate('xpack.infra.hostsViewPage.table.nameColumnHeader', {
+  defaultMessage: 'Name',
+});
+
+const osLabel = i18n.translate('xpack.infra.hostsViewPage.table.operatingSystemColumnHeader', {
+  defaultMessage: 'Operating System',
+});
+
+const cpuCountLabel = i18n.translate('xpack.infra.hostsViewPage.table.numberOfCpusColumnHeader', {
+  defaultMessage: '# of CPUs',
+});
+
+const diskLatencyLabel = i18n.translate('xpack.infra.hostsViewPage.table.diskLatencyColumnHeader', {
+  defaultMessage: 'Disk Latency (avg.)',
+});
+
+const averageTXLabel = i18n.translate('xpack.infra.hostsViewPage.table.averageTxColumnHeader', {
+  defaultMessage: 'TX (avg.)',
+});
+
+const averageRXLabel = i18n.translate('xpack.infra.hostsViewPage.table.averageRxColumnHeader', {
+  defaultMessage: 'RX (avg.)',
+});
+
+const averageTotalMemoryLabel = i18n.translate(
+  'xpack.infra.hostsViewPage.table.averageMemoryTotalColumnHeader',
+  {
+    defaultMessage: 'Memory total (avg.)',
+  }
+);
+
+const averageMemoryUsageLabel = i18n.translate(
+  'xpack.infra.hostsViewPage.table.averageMemoryUsageColumnHeader',
+  {
+    defaultMessage: 'Memory usage (avg.)',
+  }
+);
+
+/**
+ * Build a table columns and items starting from the snapshot nodes.
+ */
+export const useHostsTable = (nodes: SnapshotNode[], { time }: HostTableParams) => {
+  const {
+    services: { telemetry },
+  } = useKibanaContextForPlugin();
+
+  const reportHostEntryClick = useCallback(
+    ({ name, cloudProvider }: HostNodeRow['title']) => {
+      telemetry.reportHostEntryClicked({
+        hostname: name,
+        cloud_provider: cloudProvider,
+      });
+    },
+    [telemetry]
+  );
+
+  const items = useMemo(() => buildItemsList(nodes), [nodes]);
+
+  const columns: Array<EuiBasicTableColumn<HostNodeRow>> = useMemo(
+    () => [
+      {
+        name: titleLabel,
+        field: 'title',
+        sortable: true,
+        truncateText: true,
+        render: (title: HostNodeRow['title']) => (
+          <HostsTableEntryTitle
+            title={title}
+            time={time}
+            onClick={() => reportHostEntryClick(title)}
```

Comment:
This was one of our concerns, initially, we planned to add the `data-test-subj` on two different DOM nodes, but since we can have more properties this didn't seem really scalable.
Would you say it is a good way, in our case, to write something like this:
```ts
<EuiLink
  data-test-subj="hostsView-table-entry-click"
  data-test-hostname={`hostsView-table-entry-${name}`}
  data-test-cloudprovider={`hostsView-table-entry-${cloudProvider}`}
  {...link}
>
  {name}
</EuiLink>
```

Or there is a better way to pass all the attributes we want on this click? Also, my thought was that we could add a lot of attributes in other cases.

Reply:
AFAIK, FS allows you to specify custom attributes to be captured and indexed in their events: https://help.fullstory.com/hc/en-us/articles/360052608053/#h_01EH89PXXQNM101STZGWKW3VMP

I wonder if we could do something like `data-analytics-*`. Then, for this purpose, we could even go with something like `data-analytics-hostsview-hostname` and `data-analytics-hostsview-cloudprovider`.

But I'd like @elastic/platform-analytics to provide their thoughts on this one.

Reply:
@afharo Any news on this point? We'd like to merge this PR into 8.7 and if passing this as a custom event is not an issue we can merge this and eventually apply this practice for future events.

Reply:
I noticed you already pinged @elastic/platform-analytics on Slack. Let's wait for their input.

Reply:
@afharo what's the benefit of introducing new fields when `data-test-subj` is already consistently being used.

Reply:
@elasticelvin we'd like to understand what's the best approach for tracking more information when they should be tracked on the same interaction, like a click. Should we rely on a custom event and track all the properties we need, or is it necessary to track them all with a single attribute (`data-test-subj`)?

Reply:
Would we be keeping the same format `data-test-xyz` or is the intention to switch to `data-analytics-xyz`
I don't see any problem with this.

Reply:
So basically is just adding as many attributes as we like behind `data-test-*` right?
```ts
<EuiLink
  data-test-subj="hostsView-table-entry-click"
  data-test-hostname={`hostsView-table-entry-${name}`}
  data-test-cloudprovider={`hostsView-table-entry-${cloudProvider}`}
  {...link}
>
  {name}
</EuiLink>
```